docs/entity-editor.rst

```diff
@@ -0,0 +1,144 @@
+
+
+The relationship section is concerned with two main tasks:
+
+* Providing an **Add Relationship** button to open a Modal which acts as our relationship-editor. The relevant code for this is present in ``src/client/entity-editor/relationship-editor/relationship-editor.tsx``
+* Rendering a list of already added relationships. This is done with the help of *RelationshipList* component present in ``src/client/entity-editor/relationship-editor/relationship-section.tsx``
+
+
+We make use of ``relationshipEditorVisible`` flag to toggle the Relationship editor modal. Within the relationship editor modal, there are two fields:
+
+**Entity Select field** : The *renderEntitySelect* function deals with this field. Here ``baseEntity`` is the entity which is being edited. The ``EntitySearchFieldOption`` allows us to search for any existing entity which we would like to link to our current *baseEntity*.
+
+We can apply some additional filters to our search, so as to the optimize search results. For example, in case of a Series entity of type X, we dont need to display search results with entities which are not of the same time.
+When we select an entity from the Search results, it gets stored as ``targetEntity``.
+
+**RelationshipType Select field** : After selecting a targetEntity, we make use of a function called ``generateRelationshipSelection`` which takes our relationshipTypes object which was passed as a prop to our entity-editor, the baseEntity, and the targetEntity.
+This function then tries to generate all possible relationships which between the two entities. We can then select the Relationship Type for our entity using the RelationshipSelect field in the editor. This sets the value of ``relationship`` and ``relationsipType`` of our state.
+
+
+When we click on Add, we pass the ``relationship`` object to the following action: 
+::
+    let nextRowID = 0;
+    export function addRelationship(data: Relationship): Action {
+        return {
+            payload: {data, rowID: `n${nextRowID++}`},
+            type: ADD_RELATIONSHIP
+        };
+    }
+
+Here the data is the ``relationship`` object we passed from our Relationship Editor props. In the payload for this action, we also pass a ``rowID``.
+This is then added to the ``relationships`` object, with the ``rowId`` acting as a key for the mapping.
+
+
+As we keep on adding relationships, they are rendered as a list on the entity-editor with the help of the aforementioned RelationshipList component. We can still edit and remove these relationships from the list.
+If we click the edit button next to a particular relationship, it opens up the relationship modal with the ``relationship`` object passed as prop to the relationshipEditor.
+
+
+Submission Section
+---------------------
+When we click on submit, the entire state of the form(rootState) is sent to an appropriate *createHandler*(``{entityType}/create/handler``). If all goes well, we are redirected to the Display Page of our current entity.
```

Comment:
I guess more details here wouldn't hurt :p

It's worth pointing out that we move from front-end code to server-side code

Quick outline:
- a handler function is created with the `makeEntityCreateOrEditHandler` utility function (link to src/server/helpers/entityRouteUtils.tsx ?) adapted to the entity type
- the rootState is validated using the validators for that entity type
- the rootState is manipulated to be in a certain format by the `transformNewForm` function in each entity's route definitions file (`src/server/routes/entity/{entityType}`)
-  that formatted form is passed to the `handleCreateOrEditEntity` function which takes care of creating and modifying the required ORM models [<link to docs/ORM>] (including other entities i.e. for relationships) and persisting the changes to the database.
- If all the above steps are successful, the user is redirected to the Display Page of the newly created entity.